src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -904,7 +904,7 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
     def isStrictFP: Boolean    = !isDeferred && (hasAnnotation(ScalaStrictFPAttr) || originalOwner.isStrictFP)
     def isSerializable         = info.baseClasses.exists(p => p == SerializableClass || p == JavaSerializableClass)
     def hasBridgeAnnotation    = hasAnnotation(BridgeClass)
-    def isDeprecated           = hasAnnotation(DeprecatedAttr)
+    def isDeprecated           = hasAnnotation(DeprecatedAttr) || (isJava && hasAnnotation(JavaDeprecatedAttr))
```

Comment:
Maybe we should instead add the `scala.deprecated` annotation to the symbol to be consistent with the ClassfileParser? Like this the deprecation message that the Scala compiler shows changes between mixed and separate compilation.

See https://github.com/scala/scala/pull/8715/files#diff-3a7361fa556877c0a876fed14a513a4b

Reply:
Good point, that's not a good state of affairs.